pkg/block/fetcher.go

```diff
@@ -428,15 +428,30 @@ func NewDeduplicateFilter() *DeduplicateFilter {
 // Filter filters out duplicate blocks that can be formed
 // from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
 func (f *DeduplicateFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
-	root := NewNode(&metadata.Meta{
-		BlockMeta: tsdb.BlockMeta{
-			ULID: ulid.MustNew(uint64(0), nil),
-		},
-	})
+	var wg sync.WaitGroup
+	for _, res := range []int64{
+		int64(0), int64(5 * 60 * 1000), int64(60 * 60 * 1000),
```

Comment:
Hm, this will work. But what about doing something more generic and building metaSlice as `map[int64][]*metadata.Meta{}`? And then filterForResolution? So if we change resolutions someday it will be still fully working?